tools/sof-tplggraph.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/python3
+
+import argparse
+from tplgtool import TplgParser, TplgFormatter
+from graphviz import Digraph
+
+if __name__ == "__main__":
+    # connect nodes
+    def connect(graph, head):
+        if head["sink"] != None and type(head["sink"]) != list:
+            graph.edge(head["name"], head["sink"]["name"])
+            connect(graph, head["sink"])
+        elif head["sink"] != None:
+            for node in head["sink"]:
+                graph.edge(head["name"], node["name"])
+                connect(graph, node)
+
+    # traverse all nodes, and add them to graph
+    def init_node(graph, head):
+        # this function is used to deal with multiple connection.
+        def inner_init(graph, head):
+            if head["sink"] != None and type(head) != list:
+                graph.node(name=head["name"])
+                init_node(graph, head["sink"])
+            elif head["sink"] != None:
+                for node in head["sink"]:
+                    init_node(graph, node)
+        if type(head) != list:
+            inner_init(graph, head)
+        else :
+            for subhead in head:
+                inner_init(graph, subhead)
+
+    parser = argparse.ArgumentParser(add_help=True, formatter_class=argparse.RawTextHelpFormatter,
+            description='A Tool to Draw Topologh Graph.')
+    parser.add_argument('filename', type=str, help='tplg file name')
+    parser.add_argument('-o', '--output', type=str, default="test.gv", help="output filename for generated graph and dot source")
```

Comment:
I make this feature into a separate python file is because it has an **optional** dependency, say graphviz. we can make this feature into sof-tplgreader.py, then the optional dependency will become a prerequisite.
About the output name, sure, we can derive the picture name from tplg name.
The gv file is actually the dot source.
will do the improvement next push.

Reply:
@aiChaoSONG Please take reference:
```
try:
	from sty import fg, bg, ef, rs, Rule, Render
	CAN_COLOUR=True
except:
	CAN_COLOUR=False
```
https://github.com/thesofproject/sof/blob/6cd8acc16a7789a919bb0dc4d5b05a3a734c4a24/tools/coredumper/sof-coredump-reader.py#L33-L37

You can do a check to decide if you can enable this feature.

Reply:
I use a cmd-line parameter(-G) to control the graph feature, and make the import of graphviz local.  with this parameter, we can exit elegantly once the graph is generated. and code for pcm dump never get executed. please review again.